tests/unit/cli/storage_layout/test_storage_layout_overrides.py

```diff
@@ -48,19 +48,18 @@ def _bar():
 bar: uint256
 
 @external
-@nonreentrant("bar")
+@nonreentrant
 def public_bar():
     pass
 
 @external
-@nonreentrant("foo")
+@nonreentrant
 def public_foo3():
     pass
     """
 
     storage_layout_override = {
-        "nonreentrant.foo": {"type": "nonreentrant lock", "slot": 8},
-        "nonreentrant.bar": {"type": "nonreentrant lock", "slot": 7},
+        "$.nonreentrant_key": {"type": "nonreentrant lock", "slot": 8},
```

Comment:
Can we add here an additional paranoid test that a too-large slot override for `$.nonreentrant_key` will not overflow but raise with: https://github.com/vyperlang/vyper/blob/2150fbc50a8b95019efcc314b153059e2ef06b70/vyper/semantics/analysis/data_positions.py#L61-L65

Reply:
the storage layout override code is kind of defunct now btw, it doesn't handle modules or transient storage. i was going to fix in some follow up PR, but the priority is low since the storage layout overrides feature is not commonly used.

Reply:
https://github.com/vyperlang/vyper/pull/3769/commits/dfda2ae72da40268fa9c18b0bfd3717d7655d59a